### PR TITLE
Add fetchOneWithErrors to 2.0

### PR DESCRIPTION
**Pull request by ssanjay1**
Add fetchOneWithErrors, which functions the same as fetchOne, but adds a Result wrapper

### CODE REVIEW

File: packages/client/src/object/fetchSingle.ts [hunk 1]
```diff
@@ -57,3 +58,31 @@ export async function fetchSingle<
...
+  >
+> {
+  try {
+    const result = await fetchSingle(client, objectType, args, objectSet);
+    return { value: result as any };
+  } catch (e) {
+    if (e instanceof Error) {
+      return { error: e };
+    }
+    return { error: e as Error };
```
Comment (ericanderson) [inline packages/client/src/object/fetchSingle.ts:86]: Is this right?

Comment (ericanderson) [reply packages/client/src/object/fetchSingle.ts:86]: Should we have a wrapper function that can take a call and create a result based on if something throws?

Comment (ssanjay1) [reply packages/client/src/object/fetchSingle.ts:86]: Hmm I'm confused, I was just emulating what we did for like `fetchPageWithErrors` https://github.com/palantir/osdk-ts/blob/d183d92b3fe0b2fa8016ea94d44eebc1e21fa2fc/packages/client/src/object/fetchPage.ts#L229

Comment (ericanderson) [reply packages/client/src/object/fetchSingle.ts:86]: Okay, lets go with it for now.

**Review by ericanderson — APPROVED** [review]

**Pull request merged**